R/pkg/inst/worker/daemon.R

```diff
@@ -31,7 +31,15 @@ inputCon <- socketConnection(
     port = port, open = "rb", blocking = TRUE, timeout = connectionTimeout)
 
 while (TRUE) {
-  ready <- socketSelect(list(inputCon))
+  ready <- socketSelect(list(inputCon), timeout = 1)
+
+  # Terminate R workers in the parent process.
+  finishedChildren <- parallel:::selectChildren()
```

Comment:
Definitely for comments.

Maybe I missed your point. Children will only return their PID on exit and `parallel:::selectChildren()` will only return children PIDs that finished they work and called `parallel:::mcexit(0L)` (related test was done https://github.com/apache/spark/pull/18320#discussion_r122605437) up to my knowledge. So, even if connecting to JVM is delayed in `worker.R` and `RRunner.scala`, it won't matter.